packages/vaadin-grid/test/column.test.js

```diff
@@ -168,6 +168,29 @@ describe('column', () => {
         expect(childCountAfter).to.be.below(childCountBefore);
       });
 
+      it('should update the structure of hidden rows', async () => {
+        // Redice the size so we end up with hidden rows
+        grid.size = 0;
+
+        // Detach grid from its parent
+        const parentNode = grid.parentNode;
+        parentNode.removeChild(grid);
+
+        // Remove a column
+        column.parentNode.removeChild(column);
+
+        // Increase the size so one of the hidden rows becomes visible again
```

Comment:
That's the first time I deal with the hidden rows concept. And I'm not sure if I got the idea.

Is it related to the Virtualizer? 

I assume there is a constant number of rows, and they become hidden as you reduce size, right?

Reply:
Virtualizer (the iron-list logic inside of it) only adds more physical items to the DOM (in grid's case, the `<tr>` elements).

When you start off with an empty grid, there are 0 rows in the DOM, but as you add more data items to grid, more row elements get created to have them visually presented.

But after that, if you again reduce the number of data items, the excess row elements that were previously created are not removed, but only hidden (for later reuse).

Reply:
The updated function maintains the cell structure of those rows, for example, a grid with 3 columns:
```html
<tr>
  <td>...</td><td>...</td><td>...</td>
</tr>
```

If you remove a column, you'd expect the row structure to update as:
```html
<tr>
  <td>...</td><td>...</td>
</tr>
```

but the previous erroneous logic filtered out the hidden rows and didn't update their inner structure in this case so you ended up with excess cells on the rows when they became visible again.

Reply:
Got it. Thank you for detailed explanation!

Then I would probably suggest adding an extra expectation to make clear what content is supposed to be in the body cell before the grid is reattached.

```js
expect(getBodyCellContent(grid, 0, 0).innerText).to.equal('????'); // <------- Like this

// Re-attach the grid
parentNode.appendChild(grid);

await nextFrame();

expect(getBodyCellContent(grid, 0, 0).innerText).to.equal('cell');
```

Reply:
There aren't really any expectations about the inner structure of a detached grid. Theoretically, the grid's inner DOM could be empty while it's detached as long as it's intact after reattaching.

Reply:
Yeah, I was quite wrong. But the current expectation is still not obvious to me.

Considering your example:
```
<tr> <td>...</td><td>...</td><td>...</td> </tr>
```
becomes:
```
<tr> <td>...</td><td>...</td> </tr>
```

If I got it correctly, there finally is changed the number of columns. So, wouldn't it be better to check the final number of columns?

Reply:
Good point, changed the test case to assert that the hidden row has the same amount of cells as the non-hidden one.
Didn't use column count for this because it isn't always 1:1 with the row cell count (hidden columns, row details cell)

Reply:
Now this assertion looks clear to me, thanks!